src/js/wrap.cpp

```diff
@@ -81,6 +82,30 @@ namespace ccf::js
     }
   }
 
+  struct UntrustedHostTime
+  {
+    std::chrono::microseconds start_time;
+    std::chrono::seconds max_execution_time;
+  };
+
+  static int js_custom_interrupt_handler(JSRuntime* rt, void* opaque)
+  {
+    UntrustedHostTime* time = reinterpret_cast<UntrustedHostTime*>(opaque);
+    auto now = ccf::get_enclave_time();
+    auto elapsed_time = now - time->start_time;
+    auto elapsed_seconds =
+      std::chrono::duration_cast<std::chrono::seconds>(elapsed_time);
+    if (elapsed_seconds.count() >= time->max_execution_time.count())
+    {
+      LOG_DEBUG_FMT("JS execution has timed out");
+      return 1;
```

Comment:
When we return 1 here, what happens next? Does the stack return to the original call to `JS_Call` with some error value? We need to eventually convert this into a HTTP error that's reported to the caller.

Reply:
It issues a `SIGTERM`. So I think will need a handler for that.

Reply:
I don't think that will work - if it tries to issue a `SIGTERM` within the enclave, it'll need to OCALL. And that'll only be handled by the host - it won't immediately terminate this JS execution. But I can't see any signal handling/sending code in `quickjs.c`, so I don't think this is how it's reporting interrupts.